rcl/src/rcl/wait.c

```diff
@@ -75,27 +80,7 @@ static void
 __wait_set_clean_up(rcl_wait_set_t * wait_set, rcl_allocator_t allocator)
 {
   if (wait_set->subscriptions) {
-    rcl_ret_t ret = rcl_wait_set_resize_subscriptions(wait_set, 0);
-    (void)ret;  // NO LINT
-    assert(ret == RCL_RET_OK);  // Defensive, shouldn't fail with size 0.
-  }
-  if (wait_set->guard_conditions) {
-    rcl_ret_t ret = rcl_wait_set_resize_guard_conditions(wait_set, 0);
-    (void)ret;  // NO LINT
-    assert(ret == RCL_RET_OK);  // Defensive, shouldn't fail with size 0.
-  }
-  if (wait_set->timers) {
-    rcl_ret_t ret = rcl_wait_set_resize_timers(wait_set, 0);
-    (void)ret;  // NO LINT
-    assert(ret == RCL_RET_OK);  // Defensive, shouldn't fail with size 0.
-  }
-  if (wait_set->clients) {
-    rcl_ret_t ret = rcl_wait_set_resize_clients(wait_set, 0);
-    (void)ret;  // NO LINT
-    assert(ret == RCL_RET_OK);  // Defensive, shouldn't fail with size 0.
-  }
-  if (wait_set->services) {
-    rcl_ret_t ret = rcl_wait_set_resize_services(wait_set, 0);
+    rcl_ret_t ret = rcl_wait_set_resize(wait_set, 0u, 0u, 0u, 0u, 0u);
```

Comment:
Are the `u`s actually required? I would expect it to build without warnings with plain `0`s.

Same below.

Reply:
It probably doesn't warn. I think there is a signed to unsigned conversion without the `u`, but I'll remove it and see if a compiler complains.

Reply:
Seems to be warning free even with `-Wconversion`